### PR TITLE
doc: Add systemd-resolved description to our DNS reference

### PR DESCRIPTION
**Pull request by cfergeau**
This will need to be iterated on after getting some Kevin's feedback, something similar would be needed for the remote configuration, ...

### CODE REVIEW

File: docs/source/topics/ref_dns-configuration.adoc [hunk 1]
```diff
@@ -17,7 +17,29 @@ Additional checks are done to verify DNS is properly configured when running [co
 [id="dns-configuration-linux_{context}"]
 == Linux
 
-On Linux, {prod} expects the following DNS configuration:
+On Linux, depending on your distribution, {prod} expects the following DNS configuration.
```
Comment (robin-owen) [inline docs/source/topics/ref_dns-configuration.adoc:20]: This sentence should end with a colon. :)

File: docs/source/topics/ref_dns-configuration.adoc [hunk 2]
```diff
@@ -17,7 +17,29 @@ Additional checks are done to verify DNS is properly configured when running [co
...
+
+=== NetworkManager + systemd-resolved
+
+This configuration is used on Fedora 33 or newer, and on Ubuntu Desktop editions.
+
+* {prod} expects NetworkManager to manage networking.
+* {prod} configures systemd-resolved  to forward requests for the `testing` domain to the `192.168.130.11` DNS server. `192.168.130.11` is the IP of the {prod} virtual machine.
```
Comment (robin-owen) [inline docs/source/topics/ref_dns-configuration.adoc:27]: `systemd-resolved` should be formatted. This goes for the following line as well. :)

Also, could you please put each sentence on its own line? Thank you!

File: docs/source/topics/ref_dns-configuration.adoc [hunk 3]
```diff
@@ -17,7 +17,29 @@ Additional checks are done to verify DNS is properly configured when running [co
...
+        resolvectl dns "$1" 192.168.130.11
+        resolvectl default-route "$1" false
+fi
+
+exit 0
+----
+
+=== NetworkManager + dnsmasq
+
+This configuration is used on Fedora 32 or older, on {rhel} and on {centos}.
```
Comment (robin-owen) [inline docs/source/topics/ref_dns-configuration.adoc:42]: Missing a comma: "`This configuration is used on Fedora 32 or older, on {rhel}, and on {centos}.`"

**Review by robin-owen — COMMENTED** [review]
Overall, LGTM -- I just see a few minor nitpicks I'd like to see fixed before approving. :)

Thank you! :+1:

**Comment by cfergeau** [comment]
I've made a few more changes than what was requested to align this with https://github.com/code-ready/crc/pull/1819

File: docs/source/topics/ref_dns-configuration.adoc (hunk 2 above)
Comment (cfergeau) [reply docs/source/topics/ref_dns-configuration.adoc]: `systemd-resolved` needs formatting, but not `NetworkManager`? I've made the change already, but looking back this seems inconsistent :)

Comment (robin-owen) [reply docs/source/topics/ref_dns-configuration.adoc]: NetworkManager is the name of the project, so we don't format it. You might have a point regarding systemd-resolved and whether we should be formatting that or not -- I'm not sure!

**Review by robin-owen — APPROVED** [review]

**Pull request merged**